### PR TITLE
[ECTM-80] QA Service + mocked miner

### PR DESCRIPTION
**Pull request by mmrozek**
# Description

- mocked miner implementation
- QA service (mining blocks using mocked miner, listing pending txs)

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+version = "2.3.0"
```
Comment (mirkoAlic) [inline .scalafmt.conf:1]: =)

File: build.sbt [hunk 2]
```diff
@@ -43,15 +44,20 @@ val dep = {
     "org.scala-sbt.ipcsocket" % "ipcsocket" % "1.0.0",
     "org.bouncycastle" % "bcprov-jdk15on" % "1.59",
     "com.typesafe.scala-logging" %% "scala-logging" % "3.9.0",
-    "org.typelevel" %% "mouse" % "0.18",
+    "org.typelevel" %% "mouse" % "0.23",
```
Comment (mirkoAlic) [inline build.sbt:47]: Maybe we could start adding new Dependencies following the approach we use in the other project. (like i started doing it here: https://github.com/input-output-hk/mantis/pull/668/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR77) WDYT?

File: src/main/scala/io/iohk/ethereum/consensus/ethash/MockedMiner.scala [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import io.iohk.ethereum.consensus.ethash.MinerResponses.{MinerIsWorking, MiningError, MiningOrdered}
+import io.iohk.ethereum.consensus.ethash.MockedMiner.MineBlock
+import io.iohk.ethereum.consensus.ethash.MockedMinerProtocol.MineBlocks
+import io.iohk.ethereum.consensus.wrongConsensusArgument
+import io.iohk.ethereum.domain.{Block, Blockchain}
+import io.iohk.ethereum.nodebuilder.Node
+import io.iohk.ethereum.utils.ByteStringUtils
+import scala.concurrent.duration._
+
+class MockedMiner(
```
Comment (mirkoAlic) [inline src/main/scala/io/iohk/ethereum/consensus/ethash/MockedMiner.scala:17]: :+1:

File: src/test/scala/io/iohk/ethereum/SpecBase.scala [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import monix.eval.Task
+import monix.execution.Scheduler
+import org.scalactic.TypeCheckedTripleEquals
+import org.scalatest._
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext
+import scala.language.higherKinds
+
+trait SpecBase extends TypeCheckedTripleEquals with DiagrammedAssertions with Matchers { self: AsyncTestSuite =>
```
Comment (mirkoAlic) [inline src/test/scala/io/iohk/ethereum/SpecBase.scala:14]: :1st_place_medal:

**Comment by mirkoAlic** [comment]
Please update insomnia workspace (Yes we already have one =) )

File: build.sbt (hunk 2 above)
Comment (mmrozek) [reply build.sbt:47]: Yes, we should do that. I will do that in separate PR

**Review by mirkoAlic — APPROVED** [review]
LGTM!

File: src/main/resources/application.conf [hunk 5]
```diff
@@ -185,8 +185,8 @@ mantis {
       }
 
       # Enabled JSON-RPC APIs over the JSON-RPC endpoint
-      # Available choices are: eth, web3, net, personal, test, daedalus, iele
-      apis = "eth,web3,net,personal,daedalus,debug"
+      # Available choices are: eth, web3, net, personal, test, daedalus, iele, qa
+      apis = "eth,web3,net,personal,daedalus,debug,qa"
```
Comment (kapke) [inline src/main/resources/application.conf:189]: shouldn't section about available consensus/pow algorithms be updated as well?
shouldn't config in universal be updated for sake of documentation?

**Pull request merged**